tests/Unit/TestCreation.hpp

```diff
@@ -12,43 +13,110 @@
+    construction_string.insert(0, 2, ' ');
+    construction_string =
+        boost::algorithm::replace_all_copy(construction_string, "\n", "\n  ");
+    construction_string.insert(
+        0, Options_detail::name_helper<OptionTag>::name() + ":\n");
+    return construction_string;
+  }
+};
+
+template <typename Tag>
+struct AddGroups<Tag, cpp17::void_t<typename Tag::group>> {
+  static std::string apply(const std::string& construction_string) noexcept {
+    return AddGroups<typename Tag::group>::apply(
+        AddGroups<void>::template apply<Tag>(construction_string));
+  }
+};
+}  // namespace TestCreation_detail
+
+/// \ingroup TestingFrameworkGroup
+/// The default option tag for `TestHelpers::test_creation()`,
+/// `TestHelpers::test_factory_creation()`, and
+/// `TestHelpers::test_enum_creation()`.
 template <typename T>
-struct Opt {
+struct TestCreationOpt {
   using type = T;
   static constexpr OptionString help = {"halp"};
 };
-}  // namespace TestCreation_detail
 
 /// \ingroup TestingFrameworkGroup
-/// Construct an object from a given string.  Each line in the string
-/// must be indented.
-template <typename T, typename Metavariables = NoSuchType>
+/// Construct an object from a given string.
+///
+/// The string must not contain the name of the option (specifically, the struct
+/// being created from options, which is the name of the struct by default if
+/// no `name()` function is present). For example, to create a struct named
+/// `ClassWithoutMetavariables` with an option tag named `SizeT` the following
+/// would be used:
+///
+/// \snippet Test_TestCreation.cpp size_t_argument
+///
+/// Option tags can be tested by passing them as the second template parameter.
+/// If the metavariables are required to create the class then the metavariables
+/// must be passed as the third template parameter. The default option tag is
+/// `TestCreationOpt<T>`.
+template <typename T, typename OptionTag = TestCreationOpt<T>,
+          typename Metavariables = NoSuchType>
 T test_creation(const std::string& construction_string) noexcept {
-  Options<tmpl::list<TestCreation_detail::Opt<T>>> options("");
-  options.parse("Opt:\n" + construction_string);
-  return options.template get<TestCreation_detail::Opt<T>, Metavariables>();
+  Options<tmpl::list<OptionTag>> options("");
+  options.parse(
+      TestCreation_detail::AddGroups<OptionTag>::apply(construction_string));
+  return options.template get<OptionTag, Metavariables>();
 }
 
 /// \ingroup TestingFrameworkGroup
-/// Construct a factory object from a given string.  Each line in the
-/// string must be indented.
-template <typename BaseClass, typename Metavariables = NoSuchType>
+/// Construct a factory object from a given string.
+///
+/// The string must contain the name of the option (specifically, the struct
+/// being created from options, which is the name of the struct by default if
+/// no `name()` function is present). Note that it is not the name of the base
```

Comment:
`name()` overrides don't work with factories.  Implementing it has been discussed in #1611, but nothing has happened yet.